realm/realm-library/src/main/java/io/realm/internal/OsObjectStore.java

```diff
@@ -71,6 +73,19 @@ public static boolean deleteTableForObject(SharedRealm sharedRealm, String class
         return nativeDeleteTableForObject(sharedRealm.getNativePtr(), className);
     }
 
+    /**
+     * Try to grab a exclusive lock on the given Realm file. If the lock can be acquired, the {@code runnable} will
+     * executed while the lock is hold. The lock will ensure no one else can read or write the Realm file at the same
+     * time.
+     *
+     * @param configuration to specify the realm path.
+     * @param runnable to run with lock.
+     * @return true if the lock can be acquired and the {@code runnable} has been executed.
+     */
+    public static boolean callWithLock(RealmConfiguration configuration, Runnable runnable) {
```

Comment:
What happens in case of an exception (within the Runnable) is the lock released? How are we reporting unchecked exception?

Reply:
yes, the lock will be released. 
I planned to make a test case for this but forgot it finally ... will add a test case.